### PR TITLE
Bugfix/aserver arecorder channels not reflecting actual device

### PR DESCRIPTION
**Pull request by wiccy46**
This addresses #67 

I also made some changes to the unit tests. 

- I added a mock behavior to Dummy if channels more than maxInputChannels/maxOutputChannels, raise the same OSError. 
- Remove dummy_devices arg as it doesn't seem necessary. 
- Add test cases with channels using DummyBackend.

### CODE REVIEW

File: pya/arecorder.py [hunk 1]
```diff
@@ -39,8 +39,11 @@ def __init__(self, sr=44100, bs=256, device=None, channels=None, backend=None, *
         self._record_all = True
         self.gains = np.ones(self.channels)
         self.tracks = slice(None)
-        self.device = device or self.backend.get_default_input_device_info()['index']
-        self.channels = channels or self.backend.get_default_input_device_info()['maxInputChannels']
+        self._device = device or self.backend.get_default_input_device_info()['index']
+        if channels:
```
Comment (thomas-hermann) [inline pya/arecorder.py:43]: I suggest to simplify as 
self.channels = channels or
     self.backend.get_device_info_by_index(self._device)['maxInputChannels']
(similiarly how it is done with device above....)

File: tests/test_arecorder.py [hunk 2]
```diff
@@ -75,14 +77,21 @@ def get_default_output_device_info(self):
...
+        self.backend = DummyBackend()
+        self.max_channels = self.backend.dummy_devices[0]['maxInputChannels']
+
+    def test_boot(self):
+        ar = Arecorder(backend=self.backend).boot()
+        self.assertTrue(ar.is_active)
+        ar.quit()
+        self.assertFalse(ar.is_active)
 
     @pytest.mark.xfail(reason="Test may get affect with PortAudio bug or potential unsuitable audio device.")
```
Comment (thomas-hermann) [inline tests/test_arecorder.py:92]: affect with -> affected by

File: tests/test_aserver.py [hunk 3]
```diff
@@ -26,6 +28,26 @@ def test_default_server(self):
         Aserver.shutdown_default_server()
         self.assertIsNone(s.stream)
 
+    def test_custom_channels(self):
+        ch = random.randint(1, 10)
```
Comment (thomas-hermann) [inline tests/test_aserver.py:32]: is it good practice to have test code depend on random arguments? wouldn't it be better to do the test on all possible channels? (assuming that the number is low / limited... Or test on 4 selected channels: first, in the middle, last, and beyond last (to test for invalid channel exception

**Review by thomas-hermann — CHANGES_REQUESTED** [review]
I added a couple of single line comments: 
- typo in test_arecorder.py
- questioning random channel argument in test code

**Review by thomas-hermann — CHANGES_REQUESTED** [review]
minor request - using or instead of if / else could be considered (see inline comment)

File: pya/arecorder.py (hunk 1 above)
Comment (wiccy46) [reply pya/arecorder.py]: Ok. I was thinking of not to have overly long line. But it is more consistent with others I guess

**Comment by wiccy46** [comment]
@thomas-hermann  can you check the changes

**Comment by wiccy46** [comment]
@thomas-hermann , since you requested changes, you need to approved that for procedure to proceed.

File: tests/test_aserver.py (hunk 3 above)
Comment (aleneum) [reply tests/test_aserver.py:32]: yeah, I agree that a precise testing strategy would be a better approach. If multiple scenarios with nearly the same setting should be evaluated, the test should be encapsulated and called multiple times with varying parameters (in this case: channels). Pytest fixtures could be useful (see [here](https://stackoverflow.com/a/41837463/1617563)).

**Pull request merged**